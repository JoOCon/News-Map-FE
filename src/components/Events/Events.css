.fa-map-pin {
  color: #a4480e;
  cursor: pointer; }

.fa-map-pin:hover {
  font-size: 2.5rem;
  color: #c07546; }
