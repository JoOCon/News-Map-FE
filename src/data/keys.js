export const googleClientId =
  '241779944002-qa42v6cvk3kpu8da0jimdr364vn41t1r.apps.googleusercontent.com';
