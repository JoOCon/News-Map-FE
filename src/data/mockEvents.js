export const eventsResponse = [
  {
    name: 'P!NK: BEAUTIFUL TRAUMA WORLD TOUR',
    e_id: 'G5vzZ4amDV3z0',
    url: 'https://www.altitudetickets.com/events/detail/pink-2019',
    img:
      'https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_PORTRAIT_3_2.jpg',
    date: 'Mon, Apr 1, 2019 7:30 PM',
    venue_name: 'Pepsi Center',
    address: '1000 Chopper Circle Denver, CO 80204',
    lat: '39.74724',
    lng: '-105.010166',
    distance: 0.75,
    favorite: false
  },
  {
    name: 'Childish Gambino',
    e_id: 'G5vzZ4aV5O3CO',
    url: 'https://www.altitudetickets.com/events/detail/childish-gambino',
    img:
      'https://s1.ticketm.net/dam/a/7d9/44a96192-d246-4946-b6b4-c1f7f975a7d9_816871_TABLET_LANDSCAPE_3_2.jpg',
    date: 'Tue, Dec 4, 2018 7:30 PM',
    venue_name: 'Pepsi Center',
    address: '1000 Chopper Circle Denver, CO 80204',
    lat: '39.74724',
    lng: '-105.010166',
    distance: 0.75,
    favorite: false
  }
];

export const favoriteEventResponse = {
  message:
    'Successfully added EnParejaDos - con Adrian Uribe y Consuelo Duval with id: G5vzZ48H7Q7cH'
};
