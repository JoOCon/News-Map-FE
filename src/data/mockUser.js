export const mockUser = {
  El: '117977763637801404146',
  Zi: {
    token_type: 'Bearer',
    access_token:
      'ya29.GlxBBtt0Yi5X7awZrvhwo7UAQjrOLQ_JwGQ0ZVcJhahzVb4kj_Z5NqMx4wAhouEXxtNs72jp9pG6TZ68H1nf3bgknrUoiWzhRnotssJFcYMiopKmF1oc0ejm8ZDCwQ',
    scope:
      'openid email profile https://www.googleapis.com/auth/userinfo.email https://www.googleapis.com/auth/plus.me https://www.googleapis.com/auth/userinfo.profile',
    login_hint:
      'AJDLj6JUa8yxXrhHdWRHIV0S13cArCX2skrQ7skgtwzDVxi66KXli5pV9QkfHh_W71S4j789mYsJSJTDHdBzZgGY2XexHHgGyA',
    expires_in: 3600,
    id_token:
      'eyJhbGciOiJSUzI1NiIsImtpZCI6IjcyOGY0MDE2NjUyMDc5YjllZDk5ODYxYmIwOWJhZmM1YTQ1YmFhODYiLCJ0eXAiOiJKV1QifQ.eyJpc3MiOiJhY2NvdW50cy5nb29nbGUuY29tIiwiYXpwIjoiMjQxNzc5OTQ0MDAyLXFhNDJ2NmN2azNrcHU4ZGEwamltZHIzNjR2bjQxdDFyLmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwiYXVkIjoiMjQxNzc5OTQ0MDAyLXFhNDJ2NmN2azNrcHU4ZGEwamltZHIzNjR2bjQxdDFyLmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwic3ViIjoiMTE3OTc3NzYzNjM3ODAxNDA0MTQ2IiwiZW1haWwiOiJweWtpbTA1OTFAZ21haWwuY29tIiwiZW1haWxfdmVyaWZpZWQiOnRydWUsImF0X2hhc2giOiJNVS16RFc2c3lqbEhTNDZsbW5vMUVnIiwibmFtZSI6IlBhdWwgSyIsInBpY3R1cmUiOiJodHRwczovL2xoNC5nb29nbGV1c2VyY29udGVudC5jb20vLXE3OWJ2NE5jX2RZL0FBQUFBQUFBQUFJL0FBQUFBQUFBQUFBL0FCdE5sYkRPMEIyU3A5U3pPWGVxZ0sxOWJvcXNUaUlWQUEvczk2LWMvcGhvdG8uanBnIiwiZ2l2ZW5fbmFtZSI6IlBhdWwiLCJmYW1pbHlfbmFtZSI6IksiLCJsb2NhbGUiOiJlbiIsImlhdCI6MTU0MDUwODUwMCwiZXhwIjoxNTQwNTEyMTAwLCJqdGkiOiI2ZTYyNTg0Mzc2NTYwNDMwNjA1Y2MxYmVlNTUxOTE5ZmNjNTVmN2VmIn0.HFfyTNUGKeLor17A-qWqSRNxfITUysVYoPwC2UFzbn06Ulx4GEj7NDm3qMXFQCxXO50dGvOcsF8MKvB2fsmRkdVZeoin5QCdZa43xbfxoTmUCPX-zKFNUt8Ln3rcewNuQubkZ15Tq1BvaIh-vOQLxGSi7_yZXhTnnGLz1QNYaIZkdMUILJkGlOqigCN68GgumUWePIWn73Z-OGeKV0AStyLWpXRYtO6gIv1o8H_XV1WT2Iud-sTINDKWSK9iuomoFEnAHRcxbOQDX9A9yHHv7ZxIyzzKTd1QQmxnB_Aiux0mxu_zsZiZU5J9LS24QV5_EaAgWVKj3nCLYytEOGbjJQ',
    session_state: { extraQueryParams: { authuser: '0' } },
    first_issued_at: 1540508499764,
    expires_at: 1540512099764,
    idpId: 'google'
  },
  w3: {
    Eea: '117977763637801404146',
    ig: 'Paul K',
    ofa: 'Paul',
    wea: 'K',
    Paa:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/ABtNlbDO0B2Sp9SzOXeqgK19boqsTiIVAA/s96-c/photo.jpg',
    U3: 'pykim0591@gmail.com'
  },
  googleId: '117977763637801404146',
  tokenObj: {
    token_type: 'Bearer',
    access_token:
      'ya29.GlxBBtt0Yi5X7awZrvhwo7UAQjrOLQ_JwGQ0ZVcJhahzVb4kj_Z5NqMx4wAhouEXxtNs72jp9pG6TZ68H1nf3bgknrUoiWzhRnotssJFcYMiopKmF1oc0ejm8ZDCwQ',
    scope:
      'openid email profile https://www.googleapis.com/auth/userinfo.email https://www.googleapis.com/auth/plus.me https://www.googleapis.com/auth/userinfo.profile',
    login_hint:
      'AJDLj6JUa8yxXrhHdWRHIV0S13cArCX2skrQ7skgtwzDVxi66KXli5pV9QkfHh_W71S4j789mYsJSJTDHdBzZgGY2XexHHgGyA',
    expires_in: 3600,
    id_token:
      'eyJhbGciOiJSUzI1NiIsImtpZCI6IjcyOGY0MDE2NjUyMDc5YjllZDk5ODYxYmIwOWJhZmM1YTQ1YmFhODYiLCJ0eXAiOiJKV1QifQ.eyJpc3MiOiJhY2NvdW50cy5nb29nbGUuY29tIiwiYXpwIjoiMjQxNzc5OTQ0MDAyLXFhNDJ2NmN2azNrcHU4ZGEwamltZHIzNjR2bjQxdDFyLmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwiYXVkIjoiMjQxNzc5OTQ0MDAyLXFhNDJ2NmN2azNrcHU4ZGEwamltZHIzNjR2bjQxdDFyLmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwic3ViIjoiMTE3OTc3NzYzNjM3ODAxNDA0MTQ2IiwiZW1haWwiOiJweWtpbTA1OTFAZ21haWwuY29tIiwiZW1haWxfdmVyaWZpZWQiOnRydWUsImF0X2hhc2giOiJNVS16RFc2c3lqbEhTNDZsbW5vMUVnIiwibmFtZSI6IlBhdWwgSyIsInBpY3R1cmUiOiJodHRwczovL2xoNC5nb29nbGV1c2VyY29udGVudC5jb20vLXE3OWJ2NE5jX2RZL0FBQUFBQUFBQUFJL0FBQUFBQUFBQUFBL0FCdE5sYkRPMEIyU3A5U3pPWGVxZ0sxOWJvcXNUaUlWQUEvczk2LWMvcGhvdG8uanBnIiwiZ2l2ZW5fbmFtZSI6IlBhdWwiLCJmYW1pbHlfbmFtZSI6IksiLCJsb2NhbGUiOiJlbiIsImlhdCI6MTU0MDUwODUwMCwiZXhwIjoxNTQwNTEyMTAwLCJqdGkiOiI2ZTYyNTg0Mzc2NTYwNDMwNjA1Y2MxYmVlNTUxOTE5ZmNjNTVmN2VmIn0.HFfyTNUGKeLor17A-qWqSRNxfITUysVYoPwC2UFzbn06Ulx4GEj7NDm3qMXFQCxXO50dGvOcsF8MKvB2fsmRkdVZeoin5QCdZa43xbfxoTmUCPX-zKFNUt8Ln3rcewNuQubkZ15Tq1BvaIh-vOQLxGSi7_yZXhTnnGLz1QNYaIZkdMUILJkGlOqigCN68GgumUWePIWn73Z-OGeKV0AStyLWpXRYtO6gIv1o8H_XV1WT2Iud-sTINDKWSK9iuomoFEnAHRcxbOQDX9A9yHHv7ZxIyzzKTd1QQmxnB_Aiux0mxu_zsZiZU5J9LS24QV5_EaAgWVKj3nCLYytEOGbjJQ',
    session_state: { extraQueryParams: { authuser: '0' } },
    first_issued_at: 1540508499764,
    expires_at: 1540512099764,
    idpId: 'google'
  },
  tokenId:
    'eyJhbGciOiJSUzI1NiIsImtpZCI6IjcyOGY0MDE2NjUyMDc5YjllZDk5ODYxYmIwOWJhZmM1YTQ1YmFhODYiLCJ0eXAiOiJKV1QifQ.eyJpc3MiOiJhY2NvdW50cy5nb29nbGUuY29tIiwiYXpwIjoiMjQxNzc5OTQ0MDAyLXFhNDJ2NmN2azNrcHU4ZGEwamltZHIzNjR2bjQxdDFyLmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwiYXVkIjoiMjQxNzc5OTQ0MDAyLXFhNDJ2NmN2azNrcHU4ZGEwamltZHIzNjR2bjQxdDFyLmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwic3ViIjoiMTE3OTc3NzYzNjM3ODAxNDA0MTQ2IiwiZW1haWwiOiJweWtpbTA1OTFAZ21haWwuY29tIiwiZW1haWxfdmVyaWZpZWQiOnRydWUsImF0X2hhc2giOiJNVS16RFc2c3lqbEhTNDZsbW5vMUVnIiwibmFtZSI6IlBhdWwgSyIsInBpY3R1cmUiOiJodHRwczovL2xoNC5nb29nbGV1c2VyY29udGVudC5jb20vLXE3OWJ2NE5jX2RZL0FBQUFBQUFBQUFJL0FBQUFBQUFBQUFBL0FCdE5sYkRPMEIyU3A5U3pPWGVxZ0sxOWJvcXNUaUlWQUEvczk2LWMvcGhvdG8uanBnIiwiZ2l2ZW5fbmFtZSI6IlBhdWwiLCJmYW1pbHlfbmFtZSI6IksiLCJsb2NhbGUiOiJlbiIsImlhdCI6MTU0MDUwODUwMCwiZXhwIjoxNTQwNTEyMTAwLCJqdGkiOiI2ZTYyNTg0Mzc2NTYwNDMwNjA1Y2MxYmVlNTUxOTE5ZmNjNTVmN2VmIn0.HFfyTNUGKeLor17A-qWqSRNxfITUysVYoPwC2UFzbn06Ulx4GEj7NDm3qMXFQCxXO50dGvOcsF8MKvB2fsmRkdVZeoin5QCdZa43xbfxoTmUCPX-zKFNUt8Ln3rcewNuQubkZ15Tq1BvaIh-vOQLxGSi7_yZXhTnnGLz1QNYaIZkdMUILJkGlOqigCN68GgumUWePIWn73Z-OGeKV0AStyLWpXRYtO6gIv1o8H_XV1WT2Iud-sTINDKWSK9iuomoFEnAHRcxbOQDX9A9yHHv7ZxIyzzKTd1QQmxnB_Aiux0mxu_zsZiZU5J9LS24QV5_EaAgWVKj3nCLYytEOGbjJQ',
  accessToken:
    'ya29.GlxBBtt0Yi5X7awZrvhwo7UAQjrOLQ_JwGQ0ZVcJhahzVb4kj_Z5NqMx4wAhouEXxtNs72jp9pG6TZ68H1nf3bgknrUoiWzhRnotssJFcYMiopKmF1oc0ejm8ZDCwQ',
  profileObj: {
    googleId: '117977763637801404146',
    imageUrl:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/ABtNlbDO0B2Sp9SzOXeqgK19boqsTiIVAA/s96-c/photo.jpg',
    email: 'pykim0591@gmail.com',
    name: 'Paul K',
    givenName: 'Paul',
    familyName: 'K'
  }
};

export const mockStateUser = {
  first_name: 'Paul',
  last_name: 'K',
  email: 'pykim0591@gmail.com',
  gid: '117977763637801404146',
  id: 3,
  created_at: '2018-10-24T23:37:44.724Z',
  updated_at: '2018-10-24T23:37:44.724Z'
};

export const mockUserResponse = {
  created_at: '2018-10-24T23:37:44.724Z',
  email: 'pykim0591@gmail.com',
  first_name: 'Paul',
  gid: '117977763637801404146',
  id: 3,
  last_name: 'K',
  updated_at: '2018-10-24T23:37:44.724Z'
};
