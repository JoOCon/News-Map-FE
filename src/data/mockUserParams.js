export const responseGoogle1 =
  'https://event-mapper-api.herokuapp.com/api/v1/users';

export const responseGoogle2 = {
  body:
    '{"user":{"google_id":"117977763637801404146","given_name":"Paul","family_name":"K","email":"pykim0591@gmail.com"}}',
  headers: { 'Content-Type': 'application/json' },
  method: 'POST'
};

export const getEvents =
  'https://app.ticketmaster.com/discovery/v2/events.json?apikey=GddBLHQGZLJjPNWsNxOrYRAoBx8Hdqrf&geoPoint=9wjr4et3f&radius=200&size=100';

export const postUser = {
  body: '{"user":{"email":"pykim0591@gmail.com"}}',
  headers: { 'Content-Type': 'application/json' },
  method: 'POST'
};

export const setFavorite = {
  body: '{}',
  headers: { 'Content-Type': 'application/json' },
  method: 'POST'
};

export const removeEvent =
  'https://event-mapper-api.herokuapp.com/api/v1/users/undefined/events/undefined';

export const removeEventOptions = {
  headers: { 'Content-Type': 'application/json' },
  method: 'DELETE'
};
