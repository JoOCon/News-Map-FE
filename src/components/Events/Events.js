import React, { Component } from 'react';
import { Marker } from 'react-mapbox-gl';
import { connect } from 'react-redux';
import Modal from 'react-responsive-modal';

import * as apiCalls from '../../utilities/apiCalls/apiCalls';

import { EventModal } from '../EventModal/EventModal';
import { EventPopup } from '../EventPopup/EventPopup';

import './Events.css';

export class Events extends Component {
  constructor() {
    super();
    this.state = {
      targetEvent: {},
      displayPopup: false,
      displayModal: false,
      hoverMessage: ''
    };
  }

  plotEvents = () => {
    const { events } = this.props;
    return events.map((eve, index) => {
      let coordinates = [eve.lng, eve.lat];
      return (
        <Marker
          onClick={event => this.handleModalClick(event, 'open')}
          onMouseEnter={event => this.showEventInfo(event, eve)}
          onMouseLeave={this.closePopup}
          key={`event-${index}`}
          coordinates={coordinates}
          anchor="bottom"
        >
          <img
            className="event-marker"
            alt="map marker for an event"
            src="http://landon-homes.net/wp-content/uploads/2015/04/map-pin.png"
          />
        </Marker>
      );
    });
  };

  showEventInfo = (event, targetEvent) => {
    event.preventDefault();
    this.setState({ targetEvent, displayPopup: true });
  };

  closePopup = () => {
    this.setState({ displayPopup: false });
  };

  handleModalClick = (event, order) => {
    event.preventDefault();
    order === 'open'
      ? this.setState({ displayModal: true })
      : this.setState({ displayModal: false });
  };

  handleFavoriteClick = async () => {
    const { first_name, last_name, gid, email, id } = this.props.activeUser;
    const { targetEvent } = this.state;
    let userObj = {
      given_name: first_name,
      family_name: last_name,
      google_id: gid,
      email
    };
    const favoritedEvent = { ...targetEvent, favorite: !targetEvent.favorite };
    if (!targetEvent.favorite) {
      const eventObj = { ...targetEvent };
      delete eventObj.favorite;
      const response = await apiCalls.setFavorite(userObj, eventObj, id);
      console.log(response);
    } else {
    }
    this.setState({ targetEvent: favoritedEvent });
  };

  handleHover = (event, hoverMessage) => {
    event.preventDefault();
    this.setState({ hoverMessage });
  };

  render() {
    const {
      targetEvent,
      hoverMessage,
      displayPopup,
      displayModal
    } = this.state;
    const event = this.plotEvents();
    return (
      <div>
        {event}
        {displayPopup && <EventPopup targetEvent={targetEvent} />}
        <Modal
          open={displayModal}
          onClose={event => this.handleModalClick(event)}
          center
        >
          <EventModal
            handleHover={this.handleHover}
            hoverMessage={hoverMessage}
            handleFavoriteClick={this.handleFavoriteClick}
            targetEvent={targetEvent}
          />
        </Modal>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  activeUser: state.activeUser
});

const mapDispatchToProps = () => ({});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Events);
