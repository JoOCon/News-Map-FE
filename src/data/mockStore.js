import * as user from './mockUser';
import * as events from './mockEvents';

export const mockStore = {
  activeUser: user.mockStateUser,
  userLocation: { latitude: 39.43243, longitude: -104.98434 },
  watchList: events.userWatchlist,
  events: events.eventsResponse
};
