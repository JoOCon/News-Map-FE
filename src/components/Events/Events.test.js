import React from 'react';
import { Events } from './Events';
import * as events from '../../data/mockEvents';
import * as user from '../../data/mockUser';

jest.mock('mapbox-gl/dist/mapbox-gl', () => ({
  Marker: () => ({})
}));

describe('Events', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Events events={events.eventsResponse} activeUser={user.mockStateUser} />
    );
  });

  it('should matchSnapshot', () => {
    expect(shallow).toMatchSnapshot();
  });

  it('should setState of displayPopup when showEventInfo is called', () => {
    const mockEvent = { preventDefault: jest.fn() };

    wrapper.instance().showEventInfo(mockEvent, {});

    expect(wrapper.state().targetEvent).toEqual({});
    expect(wrapper.state().displayPopup).toEqual(true);
  });

  it('should setState of displayPopup when closePopup is called', () => {
    wrapper.instance().closePopup();

    expect(wrapper.state().displayPopup).toEqual(false);
  });

  it('should setState of displayModal when correct params are passed to handleModalClick', () => {});
});
