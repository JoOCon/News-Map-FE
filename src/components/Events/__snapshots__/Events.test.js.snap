// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events should matchSnapshot 1`] = `[Function]`;
